conformance/lang/expressions/literals/int_literal.balt

```diff
@@ -0,0 +1,502 @@
+Labels: int
+
+public function main() {
+    int x = 0x; // @error invalid HexIntLiteral due to absence of HexDigits
+
+    x = 0X; // @error invalid HexIntLiteral due to absence of HexDigits
+}
+
+Test-Case: parser-error
+Description: Test invalid HexIntLiteral.
+Labels: int
+
+public function main() {
+    int x = 0x123g; // @error invalid HexIntLiteral due to incorrect HexDigits
+
+    x = 0X123G; // @error invalid HexIntLiteral due to incorrect HexDigits
+
+    x = 123ab; // @error undefined symbol 'ab'
+}
+
+Test-Case: output
+Description: Test DecimalNumbers with toString().
+Labels: DecimalNumber, value:toString
+
+public function main() {
+    io:println((101).toString()); // @output 101
+
+    io:println(0.toString()); // @output 0
+
+    io:println(9223372036854775807.toString()); // @output 9223372036854775807
+}
+
+Test-Case: output
+Description: Test HexIntLiteral with toString().
+Fail-Issue: ballerina-lang#32769
+Labels: HexIntLiteral, value:toString
+
+public function main() {
+    io:println(0x12ab.toString()); // @output 4779
+}
+
+Test-Case: output
+Description: Test DecimalNumbers with toBalString().
+Labels: DecimalNumber, value:toBalString
+
+public function main() {
+    io:println((101).toBalString()); // @output 101
+
+    io:println(0.toBalString()); // @output 0
+
+    io:println(9223372036854775807.toBalString()); // @output 9223372036854775807
+}
+
+Test-Case: output
+Description: Test HexIntLiteral with toBalString().
+Fail-Issue: ballerina-lang#32769
+Labels: HexIntLiteral, value:toBalString
+
+public function main() {
+    io:println(0x12ab.toBalString()); // @output 4779
+}
+
+Test-Case: output
+Description: Test DecimalNumbers with return statement.
+Labels: int, return-stmt
+
+public function main() {
+    io:println(printDecimalNumber()); // @output 5674
+}
+
+function printDecimalNumber() returns int {
+    return 5674;
+}
+
+Test-Case: output
+Description: Test HexIntLiteral with return statement.
+Labels: HexIntLiteral, int, return-stmt
+
+public function main() {
+    io:println(printHexIntLiteral()); // @output 77739
+}
+
+function printHexIntLiteral() returns int {
+    return 0x12fab;
+}
+
+Test-Case: parser-error
+Description: Test possible invalid int-literal scenarios.
+Labels: int
+
+public function main() {
+    int x = 0b10111; // @error expected a 'int', but found an undefined symbol '0b10111'
+
+    x = 123 456 789; // @error invalid expression
+
+    x = 1,123,456,789; // @error invalid expression
+
+    x = 0x 12fa; // @error invalid HexIntLiteral
+
+    x = 0h123ca; // @error expected a 'int', but found an undefined symbol '0h123ca'
+
+    x = Ox456fa; // @error expected a 'int', but found an undefined symbol 'Ox456fa'
+}
+
+Test-Case: error
+Description: Numeric value is greater than upper bound.
+Labels: DecimalNumber, decimal
+
+public function main() {
+    decimal d = 10092233720368547758081009223372036854775; // @error number '10092233720368547758081009223372036854775' is too large
```

Comment:
Let's also update the binding patterns wherever applicable to avoid unused variables.

```suggestion
    decimal _ = 10092233720368547758081009223372036854775; // @error number '10092233720368547758081009223372036854775' is too large
```